### ISSUE TITLE
Unable to setup

### ISSUE THREAD

**Issue by Mattssn** [Harha/trackmania-server-docker#7]
Hello, this could very well be me doing something wrong, but I used your docker-compose file and built a stack for the three applications in docker, but when I look at the logs, I see this 

Loading dedicated server configuration...
...ERROR: Couldn't load 'cfg_server.xml'.
...Could not load the dedicated server configuration
Listening for xml-rpc commands on port 5000.
Match settings file : MatchSettings\cfg_tracklist.xml
Loading match settings...
...ERROR: The playlist file not found.
...Could not load the match settings
Server not started: no Maps available. Check your matchsettings!
URL: uplay://launch/5595/0/#join=172.21.0.3::2350@Trackmania
Server not running, exiting.
Exiting...

Where am I supposed to see the server config file, and also if I add maps don't I somehow need to add maps to cfg_tracklist.xml

With the volumes mapped in the compose, I can't seem to find these files on the host to edit, it did create some folders that lool like they should actually be files, these are supposed to be files correct?

![image](https://user-images.githubusercontent.com/54347017/129909620-e7d100ae-f0d4-471c-874d-7114638df357.png)
![image](https://user-images.githubusercontent.com/54347017/129909719-40f67d20-0a69-428a-9631-e84b07eaeb63.png)

**Comment by Mattssn** [comment]
After further investigation, when I Exec into the container, the xml files are not actual files either, I can cd into them as they are directories.

![K03RGAl](https://user-images.githubusercontent.com/54347017/129926056-bd7bb777-b8fd-44bb-8fe9-ab4f1f5d7845.png)
![SmbpYR1](https://user-images.githubusercontent.com/54347017/129926058-a829987a-03a2-4f6c-ad49-ff9f9404a5a6.png)

**Comment by Harha** [comment]
How did you deploy the compose file? You will obviously need these files on the machine that you're using to deploy the compose configuration.

So eg. you're doing ´docker-compose -p tm_server -f docker-compose.yaml up -d´ - You obviously must have file `cfg_server.xml` on the same directory you're running that ´docker-compose up´ command from, and so on...

**Comment by Mattssn** [comment]
Ah, for some reason I thought this deployed the config and tracklist file default files you have in the repo. that's my mistake, so I should erase all the files pre-create the the directories on the host with the those files, make sure everything points to the right directories in the compose volume section and docker-compose up?

**Comment by Mattssn** [comment]
Another quick question, is there something I can add to the compose to make everything run under my user and not root, I tried PUID and PGID under each service, that didn't work, all of my other docker containers are under my username not root.

**Comment by Harha** [comment]
@mattssn11 
The configuration files that are specified in the compose file on volume sections are meant to be created on the machine that deploys the compose configuration. So if you are using my example compose file as a starting point, just have the files as they are in this repository in the ./compose -folder.

So yes, you must have `./cfg_server.xml` on the host and `./pyplanet/settings` on the host in appropriate folder relative to the docker-compose deployment directory and so on. They are bind mounts, meaning if you edit/delete the file on the host, the change is reflected in the container.

The docker images itself do not contain these mandatory configuration files as-is.

**Comment by Harha** [comment]
> 
> 
> Another quick question, is there something I can add to the compose to make everything run under my user and not root, I tried PUID and PGID under each service, that didn't work, all of my other docker containers are under my username not root.

You should check out podman and see if you can get it to work. I haven't tested these images on anything else except linux 64bit docker though.

**Comment by Mattssn** [comment]
> 
> 
> > Another quick question, is there something I can add to the compose to make everything run under my user and not root, I tried PUID and PGID under each service, that didn't work, all of my other docker containers are under my username not root.
> 
> You should check out podman and see if you can get it to work. I haven't tested these images on anything else except linux 64bit docker though.

I am on Ubuntu 18.04 I have 30 other containers that are all using docker compose, it's just a hassle to have this all setup under root, everything else uses my normal username, I just wasn't sure if there was something I could slap into the compose file to make my user own all the files, the rest of my containers use something like 

    environment:
      - VIRTUAL_HOST=organizr.redacted
      - VIRTUAL_PORT=80
      - LETSENCRYPT_HOST=organizr.redacted
      - LETSENCRYPT_EMAIL=redacted
      - PUID=1000
      - PGID=1000

I tried to add PUID: 1000 and PGID: 1000 under the environment settings for the server service to see if it would create all the files under my username but they are still under root, I know this isn't under the scope of your project, I guess I was just asking for a bit of friendly help haha, if not I can try to figure that one out on my own

**Comment by Harha** [comment]
> 
> 
> > > Another quick question, is there something I can add to the compose to make everything run under my user and not root, I tried PUID and PGID under each service, that didn't work, all of my other docker containers are under my username not root.
> > 
> > 
> > You should check out podman and see if you can get it to work. I haven't tested these images on anything else except linux 64bit docker though.
> 
> I am on Ubuntu 18.04 I have 30 other containers that are all using docker compose, it's just a hassle to have this all setup under root, everything else uses my normal username, I just wasn't sure if there was something I could slap into the compose file to make my user own all the files, the rest of my containers use something like
> 
> ```
> environment:
>   - VIRTUAL_HOST=organizr.redacted
>   - VIRTUAL_PORT=80
>   - LETSENCRYPT_HOST=organizr.redacted
>   - LETSENCRYPT_EMAIL=redacted
>   - PUID=1000
>   - PGID=1000
> ```
> 
> I tried to add PUID: 1000 and PGID: 1000 under the environment settings for the server service to see if it would create all the files under my username but they are still under root, I know this isn't under the scope of your project, I guess I was just asking for a bit of friendly help haha, if not I can try to figure that one out on my own

Are you deploying it remotely somehow or from the host machine directly? I don't see how you couldn't deploy it under a specific user if you just run the docker-compose commands as that user.

I only mentioned podman because I thought you wanted to run the docker daemon as non-root user, that's what podman is, rootless container system. However what you explained should be possible just fine with docker.

**Comment by Mattssn** [comment]
> 
> 
> > > > Another quick question, is there something I can add to the compose to make everything run under my user and not root, I tried PUID and PGID under each service, that didn't work, all of my other docker containers are under my username not root.
> > > 
> > > 
> > > You should check out podman and see if you can get it to work. I haven't tested these images on anything else except linux 64bit docker though.
> > 
> > 
> > I am on Ubuntu 18.04 I have 30 other containers that are all using docker compose, it's just a hassle to have this all setup under root, everything else uses my normal username, I just wasn't sure if there was something I could slap into the compose file to make my user own all the files, the rest of my containers use something like
> > ```
> > environment:
> >   - VIRTUAL_HOST=organizr.redacted
> >   - VIRTUAL_PORT=80
> >   - LETSENCRYPT_HOST=organizr.redacted
> >   - LETSENCRYPT_EMAIL=redacted
> >   - PUID=1000
> >   - PGID=1000
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > I tried to add PUID: 1000 and PGID: 1000 under the environment settings for the server service to see if it would create all the files under my username but they are still under root, I know this isn't under the scope of your project, I guess I was just asking for a bit of friendly help haha, if not I can try to figure that one out on my own
> 
> Are you deploying it remotely somehow or from the host machine directly? I don't see how you couldn't deploy it under a specific user if you just run the docker-compose commands as that user.
> 
> I only mentioned podman because I thought you wanted to run the docker daemon as non-root user, that's what podman is, rootless container system. However what you explained should be possible just fine with docker.

Yeah everything else has been fine, for all my other containers it creates the volume mounts under my user, the docker compose file for this is under my user and I am running the docker compose command from my user, no sudo or anything

![image](https://user-images.githubusercontent.com/54347017/129936113-76f5f633-f371-43e5-af5a-8477d31f798a.png)

I did change the compose to mount the files in /opt/trackmania but that should not have made much of a difference here. its not the end of the world, I was just wondering if there was a way for me to make it all uniform.

Edit: Just to confirm, I am deploying it directly from the host machine.

**Comment by Harha** [comment]
Well I honestly have no idea what could cause that, wish you luck.

**Comment by Mattssn** [comment]
Thanks, just going to set it up under root and call it a day, thanks a ton for your help, and thanks for dockerizing this!

**Comment by Mattssn** [comment]
Actually, please don't hate me, I figured I would ask here instead of opening a new topic, got the server to start, I was able to connect to it in game, sot that part is fine, trying tog et pyplanet working, I am getting a repeated error in the logs.

INFO     [default][pyplanet.core.gbx.remote] Couldn't connect to Dedicated Server. Retry 1 of 10 (Error: [Errno 111] Connect call failed ('172.21.0.3', 5005)
INFO     [default][pyplanet.core.gbx.remote] Couldn't connect to Dedicated Server. Retry 2 of 10 (Error: [Errno 111] Connect call failed ('172.21.0.3', 5005)

I have verified this is the correct ip assigned in docker, and I do have the expose 5005 in the compose, I am not sure why its not connecting, any quick insight?

**Comment by Harha** [comment]
Why would I hate you? :D Anyways, did you get it working?

I'm sorry but I've been sleeping and then working during the day so I haven't had time to respond.

**Comment by Mattssn** [comment]
> 
> 
> Why would I hate you? :D Anyways, did you get it working?
> 
> I'm sorry but I've been sleeping and then working during the day so I haven't had time to respond.

No problem, I appreciate your help with all of this, truly.

I did get it working, I was not thinking and copied my config from the local instance I was testing on, after hours of not being able I figured out what was wrong. 

<xmlrpc_allowremote>True</xmlrpc_allowremote> was set to false haha. Hours of troubleshooting for such an easy fix! but its all working now, though I was looking at EvoSC controller, might play around with that, Pyplanet is pretty great though so not sure which one I will stick with.
